src/realm/array_string.cpp

```diff
@@ -340,7 +340,7 @@ void ArrayString::clear()
     }
 }
 
-size_t ArrayString::find_first(StringData value, size_t begin, size_t end) const noexcept
+size_t ArrayString::find_first(StringData value, size_t begin, size_t end, std::optional<StringID> id) const noexcept
```

Comment:
Maybe you can pass the optional String ID by const ref?  ... I guess it doesn't matter too much though, since StringID should be an integer right?

Reply:
It should not make any difference.